### PR TITLE
[SN1-675] Feat/balance model validation mechanism

### PR DESCRIPTION
**Pull request by ronnie-samaroo**

### CODE REVIEW

File: insights/protocol.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 MODEL_TYPE_FUNDS_FLOW = "funds_flow"
 MODEL_TYPE_FUNDS_FLOW_ID = 2
 
+MODEL_TYPE_BALANCE_TRACKING = "balance_tracking"
+MODEL_TYPE_BALANCE_TRACKING_ID = 3
```
Comment (aph5nt) [inline insights/protocol.py:14]: will we need model ids ? those can be now obsolete, same for type? please investigate

File: insights/protocol.py [hunk 2]
```diff
@@ -77,25 +80,26 @@ def get_network_id(network):
 
 def get_model_id(model_type):
```
Comment (aph5nt) [inline insights/protocol.py:81]: do we need this too then?

File: insights/protocol.py [hunk 3]
```diff
@@ -77,25 +80,26 @@ def get_network_id(network):
...
-    model_type: str = None
-    graph_schema: Optional[Dict] = None
+    model_types: List[str] = None
     #TODO: implement method for getting graph schema from miner
 
 class DiscoveryOutput(BaseModel):
     metadata: DiscoveryMetadata = None
     block_height: int = None
     start_block_height: int = None
+    balance_model_last_block: int = None
```
Comment (aph5nt) [inline insights/protocol.py:102]: idea: lets add funds_flow_model_last_block and keep block_height for compatibility ; please verify if that does make sense

File: insights/protocol.py [hunk 4]
```diff
@@ -131,12 +135,16 @@ def deserialize(self) -> Dict:
         return self.output
 
 class Challenge(BaseSynapse):
+    model_type: str # model type
```
Comment (aph5nt) [inline insights/protocol.py:138]: maybe we should add here challenge type? then we can reuse MODEL_TYPE_BALANCE_TRACKING and MODEL_TYPE_FUNDS_FLOW

File: neurons/miners/miner.py [hunk 5]
```diff
@@ -46,10 +47,10 @@ def get_config():
             help="Set miner's supported blockchain network.",
         )
         parser.add_argument(
```
Comment (aph5nt) [inline neurons/miners/miner.py:49]: is this revelant ?

**Review by aph5nt — CHANGES_REQUESTED** [review]

File: insights/protocol.py (hunk 3 above)
Comment (ronnie-samaroo) [reply insights/protocol.py:102]: Your idea is to have like the following?
```
class DiscoveryOutput(BaseModel):
    metadata: DiscoveryMetadata = None
    block_height: int = None
    funds_flow_model_last_block: int = None
    funds_flow_model_start_block: int = None
    balance_model_last_block: int = None
    run_id: str = None
    version: Optional[int] = VERSION
```

File: insights/protocol.py (hunk 1 above)
Comment (ronnie-samaroo) [reply insights/protocol.py:14]: it can be removed

File: insights/protocol.py (hunk 3 above)
Comment (aph5nt) [reply insights/protocol.py:91]: balance_model_last_block, but where is balance_model_start_block ?

Comment (aph5nt) [reply insights/protocol.py:91]: or we assume that starting block for balance tracking will be always 1 ?

**Review by aph5nt — APPROVED** [review]

**Review by wendellswa06 — APPROVED** [review]

**Pull request merged**